### DISCUSSION TITLE
Dual column display on tablet?

### DISCUSSION THREAD

**Discussion by Nuero987** [breezy-weather/breezy-weather#1306] [Ideas]
### Describe your idea

Dual column display cards and hide location list in landscape mode on tablets maybe has a good appearance.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
1. Could you make a quick mockup so I understand exactly what you're thinking of (mix of 1-column and 2 columns à la Apple Weather, or 1 column for all cards gallery-like, or 1 column for all cards equally side-by-side, or something else)? It doesn't have to be something fancy, just aggregating some parts of screenshots together is fine, as long as you keep the proportions correct.

2. Are you planning to make a pull request to implement it?

**Reply by Nuero987** [reply]
I just think there's too much space on either side in landscape.
![Screenshot_20240915-2322134](https://github.com/user-attachments/assets/dab24a25-cec6-4650-8848-b495a6c2622d)
Apple Weather for iPad is a good idea but card's background color needs to be considered. The effect will be very bad if the card takes up too much of the screen. (just like a lot of sticky notes on the screen.)
![Screenshot_20240915-232213](https://github.com/user-attachments/assets/a519c17a-69d4-4270-8c24-69f80a2419d4)
How do you feel about half fixed and the other half for cards sliding?

**Reply by papjul** [reply]
> I just think there's too much space on either side in landscape.

This empty space is supposed to be covered in great part by the location list.

> How do you feel about half fixed and the other half for cards sliding?

What would be the added value of having current info fixed? What would happen to the main screen when you open the location list?

**Reply by Nuero987** [reply]
> What would be the added value of having current info fixed? 

I don't think the location list has to take up a certain space. For people who only care about the weather where they live, displaying more information instead of location list can improve efficiency.

> What would happen to the main screen when you open the location list?

Narrow the distance between cards like in portrait mode. For devices with a small aspect ratio (folding screen), maybe we can push it back to the top of the cards.

Of course, I think my arrangement still has something inappropriately. There should be a better landscape mode effect.

**Reply by papjul** [reply]
> > What would be the added value of having current info fixed?
> 
> I don't think the location list has to take up a certain space. For people who only care about the weather where they live, displaying more information instead of location list can improve efficiency.

Maybe the issue in that case is that you're holding your tablet in landscape mode? I don't think you would have more info compared to when holding the tablet in portrait?

> > What would happen to the main screen when you open the location list?
> 
> Narrow the distance between cards like in portrait mode. For devices with a small aspect ratio (folding screen), maybe we can push it back to the top of the cards.
> 
> Of course, I think my arrangement still has something inappropriately. There should be a better landscape mode effect.

It adds another layer of complexity for a very specific case (tablet + landscape + location list hidden).

Currently, the only tablet specificity is the location list drawer, which is a nice addition from the previous developer. It fits nicely, while I'm not entirely convinced by what you suggest and would be more in favor of a tablet-specific design in that case.

Given we don't actively maintain the tablet (or TV) modes, I'm not sure of wanting a tablet-specific design merged in, that would be kind of a new app to maintain. #863 may be a way to arrange your own design by having an easy way to re-use our data.

Before we start making unrealistic plans, could you answer the previous question:

> Are you planning to make a pull request to implement it?

**Reply by Nuero987** [reply]
It depends. If other users have the same needs, I am willing to give it a try.

**Reply by papjul** [reply]
Your initial mockup tries too much to keep the smartphone look, could you try to make a mockup of something more tablet-like?

See for example:
![DSCF2085](https://github.com/user-attachments/assets/bcaa4379-7697-4dfc-bfe2-a4d439270b6c)

In our case, we would have the location list on the left, and on the right, the cards. The difficulty here is that the user can remove or reorder cards in the settings. I'm not sure how to make it nice like on the example above, while keeping this feature.

What do you think?
